rubyforgood/share_christmas
# Add RSpec, FactoryGirl, Capybara, PhantomJS

Closes #100 
- You need to install PhantomJS (`brew install phantomjs` on macOS with homebrew). Readme is updated with a link to instructions for other OSes
- Did not add headless gem because it wasn't necessary. If we opt for using something other than phantom in the future we may need it
- I included 2 basic tests to make sure at least the User factory was working as expected. A successful & failed sign in
- I cleaned up the Gemfile a bit, following the convention I normally follow (alphabetical within respective groups)
- Fixed the ActiveAdmin error by moving the file I had previously in the `/lib` folder into the initializer. This definitely feels like some kind of load path error but I didn't want to spend a lot of time with it. I'd suggest we remove ActiveAdmin in the future. It was fine for the demo but I wouldn't use it for a real application. We should look at cleaning up the routes & controllers if the data model is set
- I did _not_ test the Paperclip/FactoryGirl factories
- I commented out the remaining model tests because they were pending (all the other pendings were commented out). We should either a) remove these entirely, or b) make use of `xdescribe` to mark them all as pending and tackle them one at a time

To test this branch:
- Pull this branch down
- Run `bin/rspec` to run the test "suite". You should see 2 passing tests
- Edit `spec/features/sign_in_spec.rb` and change line 30 to read something like "Invalid email oooor password.' Run `bin/rspec` again. You should see:

An HTML file & screenshot were generated

![image](https://cloud.githubusercontent.com/assets/191564/16549527/2c244a6e-416e-11e6-9534-c80e7f87a16c.png)

If you look in `tmp/capybara` you should actually see that HTML file & PNG file. The PNG file should show a screenshot of what the page looks like
